store/postgres/src/store.rs

```diff
@@ -1220,6 +1220,94 @@ impl ChainStore for Store {
     }
 }
 
+impl EthereumCallCache for Store {
+    fn get_call(
+        &self,
+        contract_address: ethabi::Address,
+        encoded_call: &[u8],
+        block: EthereumBlockPointer,
+    ) -> Result<Option<Vec<u8>>, Error> {
+        use crate::db_schema::{eth_call_cache, eth_call_meta};
+        use diesel::dsl::sql;
+
+        let id = contract_call_id(contract_address, encoded_call, block);
+        let conn = &*self.get_conn()?;
+
+        if let Some((return_value, update_accessed_at)) = eth_call_cache::table
+            .find(id.as_ref())
+            .inner_join(eth_call_meta::table)
```

Comment:
The db schema is missing an index on `eth_call_cache.contract_address`; but that index would be pretty terrible, since there will be a lot of entries with the same contract address.

It might be better to get even fancier with this query by doing something like
```sql
  select return_value,
             (select current_date > accessed_at
                from eth_call_meta m
               where m.contract_address = c.contract_address)
   from eth_call_cache c
  where c.id = $id
```

That way, we avoid the join, don't need the (not very useful) index, and do all lookups on primary keys.

Reply:
I decided to not fear the join because I imagined the query as it is would be executed like the query you describe. The "nested loop join" should be the obvious choice for the planner, and it works exactly like that query, I don't think that index would be used by this join.

Reply:
We should check that that is what actually happens once we have some real-world data in these tables; but it's a reasonable assumption.